app/controllers/products_controller.rb

```diff
@@ -1,28 +1,47 @@
 class ProductsController < ApplicationController 
-    def index
-        @products= Product.all
-        @order_item = current_order.order_items.new
-    end
+	def index
+		@products= Product.all
+		@order_item = current_order.order_items.new
+	end
 
-    def new
-        @product = Product.new
-    end
+	def new
+		@product = Product.new
+	end
 
-    def create
-        @product = Product.new(product_params)
-        if @product.save
-            @product.avatar.attach(params[:product][:avatar])
-            redirect_to root_path
-        else
-            redirect_to new_product_path
-        end
+	def create
+		@product = Product.new(product_params)
+		if @product.save
+				@product.avatar.attach(params[:product][:avatar])
+				redirect_to root_path
+		else
+				redirect_to new_product_path
+		end
 
-    end
+	end
 
-    def show; end
+	def edit
+		@product = Product.find(params[:id])
```

Comment:
you can also create a `set_product` method and use it as a `before_action` callback